lib/shanty_docker_plugin/docker_file.rb

```diff
@@ -0,0 +1,48 @@
+require 'erubis'
+require 'fileutils'
+require 'shanty/env'
+require 'shanty_docker_plugin'
+
+module DockerPlugin
+  # Public: Process a dockerfile as an ERB template
+  class DockerFile
+    include Shanty::Logger
+
+    attr_reader :relative_output_file, :output_file, :contents
+
+    # Public: initialize the docker file
+    #
+    # base_dir  - the directory where the docker file is located
+    # registry  - the docker registry to inject into the template
+    # tag       - the build tag to inject into the template
+    # artifacts - an array of relative paths to artifacts built by dependencies
+    # build_dir - the directory to output the calculated dockerfile
+    def initialize(base_dir, registry, tag, build_dir, artifacts = nil)
+      @output_dir = File.join(base_dir, build_dir)
+      @relative_output_file = File.join(build_dir, DOCKER_FILE)
+      @output_file = File.join(base_dir, @relative_output_file)
+      @contents = process(File.join(base_dir, DOCKER_FILE), registry, tag, artifacts)
```

Comment:
You should try to avoid doing work in the constructor like this. Make `contents` a `private` method instead, and call it when needed in `write!`.